chain/network/src/peer_store.rs

```diff
@@ -187,21 +221,149 @@ impl PeerStore {
         store_update.commit().map_err(|err| err.into())
     }
 
-    pub fn add_peers(&mut self, peers: Vec<PeerInfo>) {
+    fn touch(&mut self, peer_id: &PeerId) -> Result<(), Box<dyn std::error::Error>> {
+        if let Some(peer_state) = self.peer_states.get(&peer_id) {
+            let mut store_update = self.store.store_update();
+            store_update.set_ser(ColPeers, &peer_id.try_to_vec()?, peer_state)?;
+            store_update.commit().map_err(|err| err.into())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Add list of peers into store.
+    /// When verified is true is because we establish direct connection with such peer and know
+    /// for sure its identity. If we receive a list of peers from another node in the network
+    /// by default all of them are unverified.
+    pub fn add_peers(
+        &mut self,
+        peers: Vec<PeerInfo>,
+        verified: VerificationLevel,
```

Comment:
I am confused here. Is it possible to have more than one peer in `peers` if `verified` is `Direct` or `Signed`?

Reply:
Rn, if verified is `Direct` or `Signed` then `peers.len() == 1` as you noticed, but didn't want to separate the logic into different function, since it is not relevant the length of `peers`.